packages/flutter_tools/lib/src/ios/ios_project_migration.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../project.dart';
+
+/// iOS project is generated from a template on Flutter project creation.
+/// Sometimes (due to behavior changes in Xcode, CocoaPods, etc) these files need to be altered
+/// from the original template.
+class IOSProjectMigration {
+  IOSProjectMigration(IosProject project, Logger logger) :
+        _xcodeProjectInfoFile = project.xcodeProjectInfoFile,
+        _podfile = project.podfile,
+        _logger = logger;
+
+  final File _xcodeProjectInfoFile;
+  final File _podfile;
+  final Logger _logger;
+
+  /// Inspect [project] for necessary migrations and rewrite files as needed.
+  void migrate() {
+    _migrateXcodeProjectInfoFile();
+    _migratePodfile();
+  }
+
+  void _migrateXcodeProjectInfoFile() {
+    if (!_xcodeProjectInfoFile.existsSync()) {
+      _logger.printTrace('Xcode project not found, skipping migration');
+      return;
+    }
+
+    _processFileLines(_xcodeProjectInfoFile, (String line) {
```

Comment:
I'm looking at this class and thinking about what happens when we want to add a second migration for something else later; it seems like it'll become very hard to maintain/understand pretty quickly.

I think it would be helpful to design this up front so that instead of a single processor, it uses an array of processors. That way we can encapsulate each conceptual migration and keep them all from bleeding together over time. It would also mean we could run individual ones selectively (see later comment). I don't think it would add too much complexity, and it would make the next migration much easier to add in a maintainable way (vs us always deciding that it's not bad enough to refactor *this* time, but maybe next time...)

Reply:
Done.